File: stack_using_array.c
#include <stdio.h>
#include <stdbool.h>
#define MAX_SIZE 101

char stack[MAX_SIZE];
int stackTop = -1;

void push(char data)
{
    stackTop = stackTop + 1;
    stack[stackTop] = data;
}

bool isEmpty()
{
    if (stackTop == -1)
        return true;
    return false;
}

char top()
{
    if (!isEmpty())
        return stack[stackTop];
}

char pop()
{
    if (!isEmpty())
    {
        char data = stack[stackTop];
        stackTop = stackTop - 1;
        return data;
    }
}

void print()
{
    for (int i = 0; i <= stackTop; i++)
    {
        printf("%d ", stack[i]);
    }
}

// int main()
// {
//     push('a');
//     push('b');
//     push('c');
//     printf("%c ", pop());
//     printf("%c ", pop());
//     printf("%c ", top());
// }

File: stack_using_array.js
let MinStack = function () {
    this.head = [];
    this.tail = -1;
}

MinStack.prototype.push = function (data) {
    this.head.push(data);
    this.tail++;
}

MinStack.prototype.pop = function () {
    this.head.pop();
    --this.tail;
    this.tail = this.tail < -1 ? -1 : this.tail;
}

MinStack.prototype.top = function(){
    return this.tail >= 0 ? this.head[this.tail] : null;
}

MinStack.prototype.getMin = function(){
    let min = Number.POSITIVE_INFINITY;
    this.head.forEach(item => {
        if(item < min)
            min = item;
    })
    return min;
}

let stack = new MinStack();
stack.push(-200);
stack.push(1);
stack.push(2);
console.log(stack.getMin());
console.log(stack.top());
stack.pop();
console.log(stack.top());